packages/core/src/websockets/utils.ts

```diff
@@ -0,0 +1,70 @@
+import { env } from '@latitude-data/env'
+import { jwtVerify, SignJWT } from 'jose'
+
+import { Result } from '../lib'
+import { TOKEN_CONFIG, TokenType, WebSocketData } from './constants'
+
+export async function generateWebsocketToken({
+  name,
+  payload,
+}: {
+  name: TokenType
+  payload: WebSocketData
+}) {
+  const isProd = env.NODE_ENV === 'production'
+  const config = TOKEN_CONFIG[name]
+  const secret = new TextEncoder().encode(env.WEBSOCKET_SECRET_TOKEN_KEY)
+  const token = await new SignJWT(payload)
+    .setProtectedHeader({ alg: 'HS256' })
+    .setExpirationTime(config.maxAge.stringValue)
+    .sign(secret)
+  return {
+    token,
+    cookiesOptions: {
+      secure: isProd,
+      domain: isProd ? `.${env.APP_DOMAIN}` : 'localhost',
```

Comment:
CORS for all the subdomains of the app domain. This way we can share cookies between web client and websockets server as auth method